src/activity_handlers.cpp

```diff
@@ -3441,10 +3443,32 @@ void activity_handlers::robot_control_do_turn( player_activity *act, player *p )
         return;
     }
 
-    // TODO: Add some kind of chance of getting the target's attention
-
-    // Allow time to pass
-    p->pause();
+    // robot can detect hacking attempt
+    // 1. roll against hacking attempt
+    // 2. if failure, then robot will
+    // 3. sound alarm, say "Remote intrusion detected! Notifying network."
+    // 4. All robots within 10 tiles that are receiving will be notified, set not_receiving
+    // 5. This robot sets not_receiving
+    /** @EFFECT_INT increases chance of successful robot reprogramming, vs difficulty */
+    /** @EFFECT_COMPUTER increases chance of successful robot reprogramming, vs difficulty */
+    const float success = p->get_skill_level( skill_id( "computer" ) ) - 1.5 * ( z->type->difficulty ) /
+                          ( ( rng( 2, p->int_cur ) / 2 ) + ( p->get_skill_level( skill_id( "computer" ) ) / 2 ) );
```

Comment:
While I'm not against this change, I think the real issue here is this formula.
This formula puts too much emphasis on the difficulty value. Here's a table of the min and max for hacking of a security bot and tank drone by intelligence and computer skill.

Security Bot - difficulty 14

0|1|5|10
-|-|-|-
1|-20,-inf|-2,-5.5|6.5,5.8
2|-20,-20|-2,-2|6.5,6.5
3|-20,-20|-2,-2|6.5,6.5
4|-20,-9.5|-2,-0.25|6.5,7
5|-20,-9.5|-2,-0.25|6.5,7
6|-20,-6|-2,0.8|6.5,7.375
7|-20,-6|-2,0.8|6.5,7.375
8|-20,-4.25|-2,1.5|6.5,7.667
9|-20,-4.25|-2,1.5|6.5,7.667
10|-20,-3.2|-2,2|6.5,7.9
12|-20,-2.5|-2,2.375|6.5,8.0909
14|-20,-2|-2,-2.667|6.5,8.25
16|-20,-1.625|-2,2.9|6.5,8.384615

Tank Drone - difficulty 126

Int|Comp 1|Comp 5|Comp 10
-|-|-|-
1|-188,-inf|-58,-89.5|-21.5,-27.8
5|-188,-93.5|-58,-42.25|-21.5,-17
10|-188,-36.8|-58,-22|-21.5,-8.9
20|-188,-17.9|-58,-10.75|-21.5,-2.6
100|-188,-2.78|-58,1.365385|-21.5,6.563636

To me, it seems as though this formula cares too much about difficulty. It's probably like the electrohack formula #29898 - old and not making a huge amount of sense.
If the goal of this PR is to address the way that the control laptop trivializes things, I think the root of the problem is here.

Reply:
I agree about the formula, it is likely the root of the problems.

I see now that I had not dug into `difficulty` so my odds were off a bit (I assumed it was just from the json, but its a calculation itself).